From a330ba3c6d8778dc0a0073dc0dd94042a1b60fda Mon Sep 17 00:00:00 2001
From: Xiaosong Wei <xiaosong.wei@intel.com>
Date: Fri, 23 Aug 2013 22:29:37 +0800
Subject: Fixed Scaling defect in BlackBorder rendering in VideoEditor

In blackborder rendering, when do scaling from 1280x720 to 176x144 or
352x288, the last two interested lines don't contain UV data. Since
Y/U/V width/height need to be even when do scaling, the output height of
AIR need to be multiple of 4 so that it can ensure YUV in area of
interest meet U/V height = 1/2 Y height.

Category: AOSP Enhancement
Domain: MEDIA
Upstream-Candidate: yes

Change-Id: I69518fb3cc3e0a5ec45aa1755f3e35b320d9dc03
Signed-off-by: Xiaosong Wei <xiaosong.wei@intel.com>
---
 .../vss/mcs/src/M4MCS_VideoPreProcessing.c         |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/libvideoeditor/vss/mcs/src/M4MCS_VideoPreProcessing.c b/libvideoeditor/vss/mcs/src/M4MCS_VideoPreProcessing.c
index 749f68e..f5a76d6 100755
--- a/libvideoeditor/vss/mcs/src/M4MCS_VideoPreProcessing.c
+++ b/libvideoeditor/vss/mcs/src/M4MCS_VideoPreProcessing.c
@@ -244,6 +244,8 @@ M4OSA_ERR M4MCS_intApplyVPP(M4VPP_Context pContext, M4VIFI_ImagePlane* pPlaneIn,
                              (M4OSA_UInt32)
                              ((pC->pPreResizeFrame->u_height * pPlaneOut->u_width)\
                              /pC->pPreResizeFrame->u_width);
+                        /*output height need to be multiple of 4 so that AIR can perform scaling correctly*/
+                        Params.m_outputSize.m_height = ((Params.m_outputSize.m_height+3) >> 2) << 2;
                         /*number of lines at the top*/
                         pImagePlanesTemp[0].u_topleft =
                              (M4MCS_ABS((M4OSA_Int32)
-- 
1.7.4.1

